tests/components/zha/test_light.py

```diff
@@ -449,119 +474,180 @@ async def async_test_zha_group_light_entity(
+    assert hass.states.get(device_3_entity_id).state == STATE_ON
+    assert hass.states.get(group_entity_id).state == STATE_ON
 
     # make the group have only 1 member and now there should be no entity
-    await zha_group.async_remove_members([device_light_2.ieee, device_light_3.ieee])
+    await zha_group.async_remove_members(
+        [GroupMember(device_light_2.ieee, 1), GroupMember(device_light_3.ieee, 1)]
+    )
     assert len(zha_group.members) == 1
-    assert hass.states.get(entity_id).state is None
+    assert hass.states.get(group_entity_id) is None
+    assert device_2_entity_id not in zha_group.member_entity_ids
+    assert device_3_entity_id not in zha_group.member_entity_ids
+
     # make sure the entity registry entry is still there
-    assert zha_gateway.ha_entity_registry.async_get(entity_id) is not None
+    assert zha_gateway.ha_entity_registry.async_get(group_entity_id) is not None
 
     # add a member back and ensure that the group entity was created again
-    await zha_group.async_add_members([device_light_3.ieee])
-    await dev3_cluster_on_off.on()
-    assert hass.states.get(entity_id).state == STATE_ON
+    await zha_group.async_add_members([GroupMember(device_light_3.ieee, 1)])
+    await send_attributes_report(hass, dev3_cluster_on_off, {0: 1})
+    await hass.async_block_till_done()
+    assert len(zha_group.members) == 2
+    assert hass.states.get(group_entity_id).state == STATE_ON
 
     # add a 3rd member and ensure we still have an entity and we track the new one
-    await dev1_cluster_on_off.off()
-    await dev3_cluster_on_off.off()
-    assert hass.states.get(entity_id).state == STATE_OFF
+    await send_attributes_report(hass, dev1_cluster_on_off, {0: 0})
+    await send_attributes_report(hass, dev3_cluster_on_off, {0: 0})
+    await hass.async_block_till_done()
+    assert hass.states.get(group_entity_id).state == STATE_OFF
+
     # this will test that _reprobe_group is used correctly
-    await zha_group.async_add_members([device_light_2.ieee])
-    await dev2_cluster_on_off.on()
-    assert hass.states.get(entity_id).state == STATE_ON
+    await zha_group.async_add_members(
+        [GroupMember(device_light_2.ieee, 1), GroupMember(coordinator.ieee, 1)]
+    )
+    await send_attributes_report(hass, dev2_cluster_on_off, {0: 1})
+    await hass.async_block_till_done()
+    assert len(zha_group.members) == 4
+    assert hass.states.get(group_entity_id).state == STATE_ON
+
+    await zha_group.async_remove_members([GroupMember(coordinator.ieee, 1)])
+    await hass.async_block_till_done()
+    assert hass.states.get(group_entity_id).state == STATE_ON
+    assert len(zha_group.members) == 3
 
     # remove the group and ensure that there is no entity and that the entity registry is cleaned up
-    assert zha_gateway.ha_entity_registry.async_get(entity_id) is not None
+    assert zha_gateway.ha_entity_registry.async_get(group_entity_id) is not None
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)